# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit/LIROps.yaml
# Commit: 3e40bdd11ab5
# Full Hash: 3e40bdd11ab5aa13a583162bb8394a234465c3a2
# Author: Lars T Hansen <lhansen@mozilla.com>
# Date: 2021-10-18 16:09:29
# Regressor Bug: 1727084
# File Overlap Count: 6
# Description:
#   Bug 1727084 - Memory64 - Ion operations. r=yury
#   
#   Add memory64 generalizations for Ion for all operations (except bulk
#   memory, done in an earlier patch).
#   
# ==============================================================================

diff -r 44b0c0bed8a6 -r 3e40bdd11ab5 js/src/jit/LIROps.yaml
--- a/js/src/jit/LIROps.yaml	Mon Oct 18 09:58:15 2021 +0000
+++ b/js/src/jit/LIROps.yaml	Mon Oct 18 09:58:16 2021 +0000
@@ -2550,6 +2550,12 @@
     base: WordSized
   mir_op: true
 
+- name: WasmAddOffset64
+  result_type: Int64
+  operands:
+    base: Int64
+  mir_op: WasmAddOffset
+
 - name: WasmBoundsCheck
   result_type: WordSized
   operands:
@@ -2577,6 +2583,11 @@
     ptr: WordSized
   mir_op: true
 
+- name: WasmAlignmentCheck64
+  operands:
+    ptr: Int64
+  mir_op: WasmAlignmentCheck
+
 - name: WasmLoadTls
   result_type: WordSized
   operands: